R/as-data-frame.R

```diff
@@ -44,38 +44,62 @@ as.data.frame.CrunchDataset <- function (x, row.names = NULL, optional = FALSE,
 }
 
 #' @rdname dataset-to-R
+#' @importFrom utils read.csv
 #' @export
 as.data.frame.CrunchDataFrame <- function (x, row.names = NULL, optional = FALSE, ...) {
     ds <- attr(x, "crunchDataset")
-    default.stringsAsFactors <- function () FALSE
-    limit <- min(c(10000, getOption("crunch.data.frame.limit")))
-    if (nrow(ds) * ncol(ds) > limit) {
-        ## TODO: switch to downloading CSV and reading that?
-        halt("Dataset too large to coerce to data.frame. ",
-            "Consider subsetting it first")
-    }
-    var_names <- names(x)
-    # TODO: something intelligent with modes
-    out <- lapply(var_names, function(var) {
-        values <- x[[var]]
-        
-        # Flatten all array variables, using only the subvariable name. We need 
-        # to create a list layer for non-dataframe values so that we can unlist 
-        # them later.
-        if (is.data.frame(values)) {
-            values <- as.list(values)
+    tmp <- tempfile()
+    write.csv(ds, tmp)
+    ds_out <- read.csv(tmp, stringsAsFactors = FALSE)
+    ds_out[] <- unlist(lapply(ds, coerceVariable, ds_out), recursive = FALSE)
```

Comment:
Is the `unlist` necessary here? It looks like from https://github.com/Crunch-io/rcrunch/pull/165/files#diff-afa1b86266079dbb588fa3c4dda30d20R83 it might not be?

Reply:
I think it's necessary because of how array variables are handled, so the `lapply` produces a list with the same length as `ds` but `ds_out` breaks all the array variables into their own columns.